control-plane/grpc/src/operations/volume/traits_snapshots.rs

```diff
@@ -0,0 +1,270 @@
+use crate::{
+    common, misc::traits::ValidateRequestTypes, operations::snapshot::SnapshotInfo, snapshot,
+    volume, volume::get_snapshots_request,
+};
+
+use stor_port::{
+    transport_api::{ReplyError, ResourceKind},
+    types::v0::transport::{Filter, ReplicaId, VolumeId},
+    IntoOption,
+};
+
+use std::{collections::HashMap, convert::TryFrom};
+use stor_port::types::v0::{
+    store::SpecStatus,
+    transport::{SnapshotId, SnapshotTxId},
+};
+
+/// Volume snapshot information.
+pub trait ICreateVolumeSnapshot: Send + Sync + std::fmt::Debug {
```

Comment:
Can we please rename this to something generic, then we can use the same for DeleteSnapshot as well. Or should we use a new one for that?

Reply:
yep, done.